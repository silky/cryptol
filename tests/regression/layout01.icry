(x where x = 1)
[x where x = 1, x where x = 2]
{ x = x where x = 1, y = y where y = 2 }
x where (x,y) = (1,2)
x where { x = x, y = y } = { x = 10, y = 20 }
