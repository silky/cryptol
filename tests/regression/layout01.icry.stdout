Loading module Cryptol
Assuming a = 1
0x1
Assuming a = 2
[0x1, 0x2]
Assuming a = 1
Assuming b = 2
{x = 0x1, y = 0x2}
[warning] at <interactive>:1:1--1:22:
  Defaulting 2nd type parameter
             of expression __p0
             at <interactive>:1:10--1:11
  to 2
[warning] at <interactive>:1:1--1:22:
  Defaulting 1st type parameter
             of expression __p0
             at <interactive>:1:12--1:13
  to 1
[warning] at <interactive>:1:1--1:22:
  Defaulting 2nd type parameter
             of expression __p0
             at <interactive>:1:12--1:13
  to 2
Assuming a = 1
0x1
[warning] at <interactive>:1:1--1:46:
  Defaulting 2nd type parameter
             of expression __p0
             at <interactive>:1:15--1:16
  to 5
[warning] at <interactive>:1:1--1:46:
  Defaulting 1st type parameter
             of expression __p0
             at <interactive>:1:22--1:23
  to 4
[warning] at <interactive>:1:1--1:46:
  Defaulting 2nd type parameter
             of expression __p0
             at <interactive>:1:22--1:23
  to 5
Assuming a = 4
0xa
